Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -928,6 +960,31 @@ public struct Timestamp: BSONValue, Equatable, Codable {
     }
 }
 
+/// A struct to represent the deprecated Undefined type.
+/// Undefined instances cannot be created, but they can be read from existing documents that contain them.
+public final class BSONUndefined: BSONValue, Equatable, Codable {
```

Comment:
I've made this a `class` because there's no point in ever making copies. `BSONNull` is essentially the same thing--should we make that a `class` as well?

Reply:
Does making it a class buy us anything? I tend to think of structs as value types, and BSONUndefined and BSONNull fit into that category.